core/src/main/java/com/datastrato/gravitino/catalog/CatalogProxyPlugin.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.catalog;
+
+import com.datastrato.gravitino.utils.Executable;
+import java.security.Principal;
+import java.util.Map;
+
+public interface CatalogProxyPlugin {
+  Object doAs(
+      Principal principal, Executable<Object, Exception> action, Map<String, String> properties)
+      throws Throwable;
```

Comment:
Can we use a more detailed exception class?

Reply:
No. We can't.